api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -200,6 +204,19 @@ type HTTP10Settings struct {
 	UseDefaultHost *bool `json:"useDefaultHost,omitempty"`
 }
 
+// HTTP2Settings provides HTTP/2 configuration on the listener.
+type HTTP2Settings struct {
+	// InitialStreamWindowSize sets the initial window size for HTTP/2 streams.
+	// If not set, the default value is 64 KiB(64*1024).
+	// +optional
+	InitialStreamWindowSize *Size `json:"initialStreamWindowSize,omitempty"`
+
+	// InitialConnectionWindowSize sets the initial window size for HTTP/2 connections.
+	// If not set, the default value is 1 MiB(1024*1024).
+	// +optional
+	InitialConnectionWindowSize *Size `json:"initialConnectionWindowSize,omitempty"`
```

Comment:
Two additional knobs that can be important here are:
- `max_concurrent_streams` - the secure-by-default value of 100 is quite strict, and users may want to increase it.
- `override_stream_error_on_invalid_http_message` - see discussion on how this field should be configured based on wether envoy is a L1 or L2 proxy: https://www.envoyproxy.io/docs/envoy/latest/configuration/best_practices/level_two. In other words, there's no "correct" value here, and we should make it configurable for users based on their topology.

Reply:
`override_stream_error_on_invalid_http_message` should be added in http1,http2,and http3 settings. Could we address this in another PR?

Reply:
created #3245